src/api/libcellml/loggererrors.h

```diff
@@ -0,0 +1,268 @@
+/*
+Copyright 2016 University of Auckland
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef LIBCELLML_LIBCELLML_LOGGERERRORS_H_
+#define LIBCELLML_LIBCELLML_LOGGERERRORS_H_
+
+#include <string>
+
+#include <libcellml/libcellml_export.h>
+#include <libcellml/types.h>
+
+namespace libcellml {
+
+/**
+ * @brief The EntityError class
+ *
+ * Base class for errors used with the logger class.
+ */
+class LIBCELLML_EXPORT EntityError
+{
+public:
+    /**
+     * @brief Set the description for this entity error.
+     *
+     * Set the @c std::string @p description for why this entity error was raised.
+     *
+     * @param description The @c std::string error description to set.
+     */
+    void setDescription(const std::string& description) { mDescription = description; }
+
+    /**
+     * @brief Get the description for this entity error.
+     *
+     * Get the @c std::string description for why this entity error was raised.
+     *
+     * @return The @c std::string description of the error.
+     */
+    std::string getDescription() const { return mDescription; }
+
+protected:
+    std::string mDescription; /**< The string description for why this entity error raised. */
+};
+
+/**
+ * @brief The NamedEntityError class
+ *
+ * NamedEntityError class for all nameable loggable errors.
+ */
+class NamedEntityError: public EntityError
```

Comment:
I imagine that you anticipate this class is going to be useful at some point and therefore decided to go ahead and have it empty for now? Whatever the case, it's not useful at the moment, so I would simply not have it. (Same with `ComponentEntityError`.)